### DISCUSSION TITLE
How to check that DatePicker is not empty?

### DISCUSSION THREAD

**Discussion by elshorbagy** [Megabit/Blazorise#4566] [Q&A] [answered]
Hi,

I have a DatePicker component and I want to check if it has a value. I tried to use `<Validation Validator="ValidationRule.IsNotEmpty">` but it kept saying that the DatePicker was empty even if there was a date value.

Thank you

**Comment by stsrki** [comment, marked as answer]
You should check the binded value, not the field or method of the `DatePicker`

**Example**

```
<DatePicker @bind-Date="@selectedValue" />
@code{
  DateTime? selectedValue;

  void SomeMethod()
  {
    if(selectedValue == null)
      // do something
  }
```

**Reply by elshorbagy** [reply]
@stsrki I want to use `<Validation Validator="ValidationRule.IsNotEmpty">`, is that possible?

**Reply by stsrki** [reply]
The `ValidationRule.IsNotEmpty` is used mostly for `string` types. If you want to check for date value types then you can create your own validation handler and use it.

`void IsNotEmpty( ValidatorEventArgs e ) => e.Status = e.Value is not DateTime ? ValidationStatus.Error : ValidationStatus.Success;`

**Comment by elshorbagy** [comment]
thank you
